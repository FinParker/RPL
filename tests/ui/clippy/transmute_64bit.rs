//@revisions: inline normal
//@ignore-bitwidth: 32
//@[inline]compile-flags: -Z mir-opt-level=0 -Z inline-mir=true
//@[normal]compile-flags: -Z mir-opt-level=0 -Z inline-mir=false

// #[rpl::dump_mir(dump_cfg, dump_ddg)]
fn main() {
    unsafe {
        let _: *const usize = std::mem::transmute(6.0f64);
        //~^ wrong_transmute

        let _: *mut usize = std::mem::transmute(6.0f64);
        //~^ wrong_transmute
    }
}
