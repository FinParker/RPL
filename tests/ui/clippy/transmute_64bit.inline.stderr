error: transmute from a `f64` to `*const usize`
  --> tests/ui/clippy/transmute_64bit.rs:9:31
   |
LL |         let _: *const usize = std::mem::transmute(6.0f64);
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[deny(rpl::wrong_transmute)]` on by default

error: transmute from a `f64` to `*mut usize`
  --> tests/ui/clippy/transmute_64bit.rs:12:29
   |
LL |         let _: *mut usize = std::mem::transmute(6.0f64);
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

